src/router/role-specific-route.vue

```diff
@@ -11,31 +15,43 @@
 </template>
 
 <script lang="ts">
-  import { Component, Vue, Prop } from 'vue-property-decorator'
+  import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
 
+  import ConfirmRouteExit from '@/mixins/confirm-route-exit'
   import store from '@/store'
   import PageNotFoundContent from '@/views/404/page-not-found-content.vue'
 
+  Component.registerHooks(['beforeRouteLeave'])
+
   @Component({
     components: {
       PageNotFoundContent
     }
   })
-  export default class ErikoistuvaRoute extends Vue {
+  export default class ErikoistuvaRoute extends Mixins(ConfirmRouteExit) {
     @Prop({ required: true })
     routeComponent!: any
 
     @Prop({ required: true, default: [] })
     allowedRoles!: string[]
 
+    @Prop({ required: false, type: Boolean, default: false })
+    confirmRouteExit!: boolean
+
     loading = true
 
     get isAllowedRoute() {
       const authorities = store.getters['auth/account'].authorities
       return this.allowedRoles.some((r) => authorities.includes(r))
     }
 
-    async mounted() {
```

Comment:
Voisi ottaa parametrina skipRouteExitConfirm arvon. Tällöin voidaan kytkeä varmistus myös takaisin päälle tietyissä tilanteissa.